share/availability/full/availability.go

```diff
@@ -0,0 +1,80 @@
+package full
+
+import (
+	"context"
+	"errors"
+
+	"github.com/celestiaorg/celestia-node/share/eds"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/share/availability/discovery"
+
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/ipfs/go-blockservice"
+)
+
+var log = logging.Logger("share/availability/full")
+
+// ShareAvailability implements share.Availability using the full data square
+// recovery technique. It is considered "full" because it is required
+// to download enough shares to fully reconstruct the data square.
+type ShareAvailability struct {
+	rtrv *eds.Retriever
+	disc *discovery.Discovery
+
+	cancel context.CancelFunc
+}
+
+// NewShareAvailability creates a new full ShareAvailability.
+func NewShareAvailability(bServ blockservice.BlockService, disc *discovery.Discovery) *ShareAvailability {
```

Comment:
Thank you, somehow `NewLightAvailability` got missed, I've fixed it to be consistent. The idea was to to leave type of availability to pkg name. So caller could call `full.NewShareAvailability` or `light.NewShareAvailability`. Since they all are implementation of same `share.Availability` interface.